src/Libraries/DSCPython/CPythonEvaluator.cs

```diff
@@ -233,7 +233,26 @@ public static DataMarshaler OutputMarshaler
                                 }
                             }
 
-                            return outputMarshaler.Marshal(pyObj.AsManagedObject(typeof(object)));
+                            var unmarshalled = pyObj.AsManagedObject(typeof(object));
+                            if (unmarshalled is PyObject)
+                            {
+                                using (unmarshalled as PyObject)
+                                {
+                                    if (unmarshalled.Equals(pyObj))
+                                    {
+                                        // Object can't be unmarshalled. Prevent a stack overflow.
+                                        throw new InvalidOperationException(Properties.Resources.FailedToUnmarshalOutput);
+                                    }
+                                    else
```

Comment:
I don't think we need this `else` or the `else` in line 252? We can simply have the `return` statement in line 254 outside of the `else`.

Reply:
Sorry, I think I misunderstood. I removed the one in line in 252, but not this one.

Conversion in line 248 happens inside the `using` statement context. The one on line 254 happens afterwards. After we dispose the variable, we shouldn't use it anymore.